tests/pytest/test_capture.py

```diff
@@ -88,12 +88,30 @@ def test_output_emitting_unicode(testdir, monkeypatch):
 TRACEBACKHIDE_TIMEOUT = """
 from hypothesis import given, settings
 from hypothesis.strategies import integers
+from hypothesis.errors import HypothesisDeprecationWarning
+
 import time
+import warnings
+import pytest
 
-@given(integers())
-@settings(timeout=1)
-def test_timeout_traceback_is_hidden(i):
-    time.sleep(1.1)
+
+# Since our tests are run with warnings-as-errors, we need to ignore
```

Comment:
I'd prefer to keep the impact of this workaround a bit more local by defining an outer test:

```python
def test_timeout_traceback_is_hidden():
    with warnings.catch_warnings(record=True):
        warnings.simplefilter('ignore', HypothesisDeprecationWarning)
        @given(integers())
        @settings(timeout=1)
        def inner(i):
            time.sleep(1.1)
        inner()
```